ft-close-incident-reports: update incident report transitional state

gitau-kinuthia opened the pull request:
#### What does this PR do?
Add functionality for closing an incident report if an asset status moves from damaged/lost to
allocated/available

#### Description of Task to be completed?
Update an assets's `transition state` to `CLOSED` when an Asset status is changed from `(DAMAGED, LOST)` to either `(AVAILABLE, ALLOCATED)`
#### How should this be manually tested?
1. On browser, goto url `{base_url}/api/v1/incidence-reports/`



#### Any background context you want to provide?


#### What are the relevant pivotal tracker stories?
[164101493](https://www.pivotaltracker.com/story/show/164101493)

#### Postman Documentation


#### Screenshots (If applicable)

gitau-kinuthia committed:
ft-close-incident-reports: update incident report transitional state

- Add model signals to listen for changes
- Add unit tests for the signals

[Delivers ##167427843]

gitau-kinuthia committed:
ft-close-incident-reports: update incident report transitional state

- update model signals
- Add unit tests for the updated signals

[Delivers #164101493]

gitau-kinuthia committed:
ft-close-incident-reports: update incident report transitional state

- Update model signals
- Add unit tests for the updated signals
- Fix failing  linting checks
- fix failing unit tests

[Delivers #164101493]

kmazi commented on core/tests/test_state_transition_model.py:
```diff
@@ -20,5 +27,149 @@ def test_create_incident_report_creates_new_state_transition(self):
+    """
+    Test updating of transition state when Asset status is modified
+    Update the transition state to CLOSED when an Asset status is changed from (DAMAGED,LOST) to
+    either (AVAILABLE,ALLOCATED)
```
It should be that transition state is CLOSED when AssetStatus changes from ALLOCATED to either DAMAGED, LOST or AVAILABLE

gitau-kinuthia replied:
Noted. The implementation has been updated 👍

kmazi commented on core/tests/test_state_transition_model.py:
```diff
@@ -20,5 +27,149 @@ def test_create_incident_report_creates_new_state_transition(self):
+        ).objects.get_or_create(asset_incident_report_id=self.incident_report.id)
+        self.transition_state = self.transition_state[0]
+
+    def test_update_asset_status_from_lost_to_available(self):
```
This is outside the context of the story. (fyi)

gitau-kinuthia replied:
Please elaborate on this

kmazi commented on core/tests/test_state_transition_model.py:
```diff
@@ -20,5 +27,149 @@ def test_create_incident_report_creates_new_state_transition(self):
+        )
+
+        # update status to lost
+        status.current_status = constants.DAMAGED
```
lol. Comment above says update status to lost. lol

gitau-kinuthia replied:
Noted and fixed

gitau-kinuthia committed:
ft-add-notifications: Add notifications model

-  Add notifications model
- add signals to create notifications
- add unit tests
- configure django_q and django_cache
- resolve merge conflicts
- update README.md
- apply PR feedback
- remove curly braces from notification body template
- update functionality to close transition report on asset status
  change to `lost` or `damaged`

[Delivers #161678177]

kmazi reviewed (commented):
Could you also add test to show that this could be accomplished via endpoint. You know, it serves as a documentation on how to use the api level interface.

gitau-kinuthia committed:
ft-add-notifications: Add notifications model

-  Add notifications model
- add signals to create notifications
- add model level unit tests
- configure django_q and django_cache
- resolve merge conflicts
- update README.md
- apply PR feedback
- remove curly braces from notification body template
- update functionality to close transition report on asset status
  change to `lost` or `damaged`
- add api level api tests

[Delivers #161678177]
